Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -0,0 +1,141 @@
+            try
+            {
+                highestBalanceAccount = _accounts[0];
+                for (int i = 1; i < _accounts.Length; i++)
+                {
+                    if (_accounts[i].GetHighestBalance() > highestBalanceAccount.GetHighestBalance())
+                    {
+                        highestBalanceAccount = _accounts[i];
+                    }
+                }
+            }
+            catch (NullReferenceException)
+            {
+                return Messages.InvalidMessage;
+            }
+            catch (IndexOutOfRangeException)
+            {
+                return Messages.InvalidMessage;
+            }
+
+            return _accounts.Length > _arrayBreak && AccountOps.AreBalancesEqual(_accounts)
+              ? $"{StringOps.FormatAndCommas(_accounts)} {Messages.HadTheMostMoneyEver}. {StringOps.FormatCurrency(highestBalanceAccount.GetHighestBalance(), _currency)}."
+              : $"{highestBalanceAccount.GetName()} {Messages.HadTheMostMoneyEver}. {StringOps.FormatCurrency(highestBalanceAccount.GetHighestBalance(), _currency)}.";
+        }
+
+        /// <summary>
+        /// Return name and loss of a person with a biggest loss (balance change negative).
+        /// </summary>
+        public string FindPersonWithBiggestLoss()
+        {
+            Account biggestLossAccount;
+            try
+            {
+                biggestLossAccount = _accounts[0];
+                for (int i = 0; i < _accounts.Length; i++)
+                {
+                    if (_accounts[i].GetLoss() < biggestLossAccount.GetLoss())
+                    {
+                        biggestLossAccount = _accounts[i];
+                    }
+                }
+            }
+            catch (NullReferenceException)
+            {
+                return Messages.InvalidMessage;
+            }
+            catch (IndexOutOfRangeException)
+            {
+                return Messages.InvalidMessage;
+            }
+
+            return _accounts.Length > _arrayBreak && AccountOps.AreBalancesEqual(_accounts)
+                ? $"{Messages.InvalidMessage}"
+                : $"{biggestLossAccount.GetName()} {Messages.LostTheMostMoney}. {StringOps.FormatCurrency(biggestLossAccount.GetLoss(), _currency)}.";
+        }
+
+        /// <summary>
+        /// Return name and current money of the richest person.
+        /// </summary>
+        public string FindRichestPerson()
+        {
+            Account richestBalanceAccount;
+            try
+            {
+                richestBalanceAccount = _accounts[0];
+                for (int i = 1; i < _accounts.Length; i++)
+                {
+                    if (_accounts[i].GetCurrentBalance() > richestBalanceAccount.GetCurrentBalance())
+                    {
+                        richestBalanceAccount = _accounts[i];
+                    }
+                }
+            }
+            catch (NullReferenceException)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Do not catch NullReferenceException; test for null instead.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5054323)